### PR TITLE
Make spacemacs/helm-smart-do-search default to symbol at point

### PR DESCRIPTION
**Pull request by justbur**
This is the default of rgrep, so this makes searching through helm (with `<SPC> s /`).

### CODE REVIEW

**Comment by syl20bnr** [comment]
I like that you want to make them consistent but some users don't like automatic insertion of symbol at point while other likes it. So here is what I propose: make a function that take the selected region so we can easily search for a symbol using `SPC v` (expand region) while keeping the control of not auto-insert anything when there is no active regions.
Also we can add a shortcut `/` in expand-region (like it is done in evil-iedit-state), so `SPC v v /` and we search directly for the selection.

**Comment by tuhdo** [comment]
I don't know why getting symbol at point is inconvenient. You can just `C-k` to discard the symbol and type new things.

**Comment by syl20bnr** [comment]
Auto-inserting a search pattern triggers a search that may be useless so it is not a good default. If we had to choose between auto-insert and not auto-insert then the good choice it to not auto-insert. The decision to auto-insert stuff should be made before the action not the other way around.

But either solution is not satisfying to me, I want to leverage expand-region which will give even more control and has the nice property of being extremely fast to type.

**Comment by justbur** [comment]
I just learned that `M-n` yanks the symbol at the point in helm, so that accomplishes what I wanted pretty easily.

I can work on your suggestion @syl20bnr. It makes sense to me. I thought of one other alternative which is to add the command to the `ahs` micro-state. So you could do something like `SPC s h` then `/` in the micro state to drop into a `helm-do-ag` with that symbol. What do you think of that?

By the way, @tuhdo `C-k` is overridden in the helm buffers to move the selection up. `C-w` also yanks from the point until the end of the symbol, so the default commands for quickly erasing a search term don't work unfortunately.

**Comment by syl20bnr** [comment]
>  The other possibility is to take the SPC s A keybinding for search for symbol at point using ag (also for pt, ack and grep). What do you think of those?

Excellent !

Integration in ahs is and expand-region (that is supporting active region) is also a must for even more flexibility (select more than a symbol).

**Comment by justbur** [comment]
Great, I just pushed the `ahs` integration, which I really like. I'll work on the rest. Do you want me to put all of this in one commit?

**Comment by syl20bnr** [comment]
The function to get the symbol at point or the active region could be in another PR.
The integration in expand-region in another PR.

**Comment by justbur** [comment]
@syl20bnr Got it. This PR is ready. I added those keybindings to the ahs commit (for `pt`, `ag` and `ack`). There doesn't seem to be an easy way to do this for `helm-do-grep` since it doesn't pick up the same configuration variable. I tried modifying the function `helm-do-grep-1` (called by `helm-do-grep`) but couldn't get it to work.

**Comment by justbur** [comment]
@syl20bnr Implementing what you seem to be suggesting for `expand-region` doesn't seem straightforward. Stopping at the symbol is easy, but once `expand-region` goes beyond a line `helm-do-ag` breaks on the newline characters. There are other hurdles. `ag`, for example, also seems to expect a pcre so having unmatched parentheses is a problem. This can be disabled in command line options, but the different search tools require different options. 

In short, `SPC v v /` is ok if you stop at the symbol (but then it's equivalent to `SPC s h /` now), but going past the symbol seems complicated. I'm willing to mess around with `helm-ag` some more if you tell me what I'm missing.

**Comment by syl20bnr** [comment]
Even if expand-region works only for the symbol it is valuable, this is base command I use to select something and then edit or surround so having the search integrated fit well this workflow.

We have control over the passed parameters for pt and ag at least, see https://github.com/syl20bnr/spacemacs/blob/develop/spacemacs/packages.el#L1678-L1684

I remember a discussion where we wanted to have control over the search format, that is, using a regexp or not which is achievable via the parameters like you said.

We could decide about one format by default and put the other one on `SPC s C-a` and `SPC s s-C-a`. I don't know what it the most useful though.

For the issue of multiple line I guess we have to find some way to convert the selected text to a regexp accepted by the tool. I would not be surprise that such functions already exists somewhere.

**Comment by syl20bnr** [comment]
Thank you ! :heart: 
Cherry-picked into develop branch, you can safely delete your branch.

PS: `SPC s G` (grep) is missing ;-)
